plugin-server/src/worker/ingestion/persons/batch-writing-person-store.ts

```diff
@@ -900,45 +897,10 @@ export class BatchWritingPersonsStoreForBatch implements PersonsStoreForBatch, B
 
             // Update the PersonUpdate with latest data and merged properties
             personUpdate.properties = mergedProperties
-            personUpdate.properties_last_updated_at = latestPerson.properties_last_updated_at || {}
-            personUpdate.properties_last_operation = latestPerson.properties_last_operation || {}
             personUpdate.version = latestPerson.version
         }
 
-        throw new Error('Assert version update failed, will retry')
-    }
-
-    private async updatePersonWithTransaction(personUpdate: PersonUpdate, source: string): Promise<TopicMessage[]> {
-        const operation = 'updatePersonTransaction' + (source ? `-${source}` : '')
-        this.incrementDatabaseOperation(operation as MethodName, personUpdate.distinct_id)
-
-        // Convert PersonUpdate back to InternalPerson for database call
-        const internalPerson = toInternalPerson(personUpdate)
-        const start = performance.now()
-
-        // Use a transaction to ensure we get the latest version with FOR UPDATE
-        const kafkaMessages = await this.db.postgres.transaction(PostgresUse.PERSONS_WRITE, operation, async (tx) => {
-            // First fetch the person with FOR UPDATE to lock the row
-            const latestPerson = await this.db.fetchPerson(personUpdate.team_id, personUpdate.distinct_id, {
-                forUpdate: true,
-            })
-
-            if (!latestPerson) {
-                throw new Error('Person not found during direct update')
-            }
-
-            // Create update object without version field (updatePerson handles version internally)
-            const { version, ...updateFields } = internalPerson
-            const [_, kafkaMessages] = await this.db.updatePerson(latestPerson, updateFields, tx, 'forUpdate')
-            return kafkaMessages
-        })
-        this.recordUpdateLatency(
-            'updatePersonWithTransaction',
-            (performance.now() - start) / 1000,
-            personUpdate.distinct_id
-        )
-        observeLatencyByVersion(internalPerson, start, operation)
-        return kafkaMessages
+        throw new RaceConditionError('Assert version update failed, will retry')
```

Comment:
question: in the case when the assert version update fails and we're going to retry, do we rely on the side-effect of updating personUpdate in place in this method? I mean, will the caller of this method assume the submitted personUpdate will be modified within the method?

If yes, could we change this to not use exceptions, but some sort of a result type?